### PR TITLE
Add GitHub Actions Lint job

### PR DESCRIPTION
**Pull request by conao3**
Add GitHub Actions Lint job.
As #2689 discussion, we need some indentation linter.

This job indent using Emacs-28 and check diff.

### CODE REVIEW

File: .github/workflows/lint.yml
Comment (ericdallo) [inline .github/workflows/lint.yml:3]: This could bother us every commit, WDYT about just running this when a PR is merged or a commit is made on master?
This way we'll do that as a cleanup only on the last commit of the pr (the merge)

Comment (conao3) [reply .github/workflows/lint.yml:3]: I think no. This linter is running for contributors, not maintainers. You, maintainers, are shared properly setting, there is no reason commits with wrong indentation.
However, we need to notice immediately that the maintainer's commits have been mixed in as well.
That's why I think this trigger is appropriate.

Comment (ericdallo) [reply .github/workflows/lint.yml:3]: This linter will run for pushes on master by maintainers as well, but my concern is more if I have a PR and push a commit, the lint will run, when I try to push another commit, it won't work, I'll need to pull the last commit, that's why I don't like CI committing on PRs until it's finished.

Running it on master pushes or PR merges would work for both contributors and maintainers

Comment (kiennq) [reply .github/workflows/lint.yml:3]: This job is not pushing anything. It's only check if there is diff with what the auto indentation and current code, and fail if there's different

Comment (conao3) [reply .github/workflows/lint.yml:3]: >  if I have a PR and push a commit, the lint will run,

yes

> when I try to push another commit, it won't work

but, really?
I think it should run too.

Comment (ericdallo) [reply .github/workflows/lint.yml:3]: Oh sorry, I thought the CI was pushing, failing when is not linted correctly is enough for me :)

Comment (conao3) [reply .github/workflows/lint.yml:3]: > This job is not pushing anything. It's only check if there is diff with what the auto indentation and current code, and fail if there's different

Yes. That is CI. You just don't merge PR if any CI fail.

Comment (ericdallo) [reply .github/workflows/lint.yml:3]: Yes, is just that I already worked on projects that the linter pushes to the branch every commit and I thought this a PR that was doing that, sorry for that

Comment (conao3) [reply .github/workflows/lint.yml:3]: No need to say sorry. Let me know if it doesn't work as I say :)

**Review by kiennq — APPROVED** [review]

**Review by ericdallo — APPROVED** [review]

File: .github/workflows/lint.yml
Comment (ericdallo) [inline .github/workflows/lint.yml:37]: I think we could move that to our Makefile, then we could lint locally, and CI just call the makefile like the other CI workflows, WDYT?

Comment (ericdallo) [reply .github/workflows/lint.yml:37]: I don't get why it needs to be specified on cask files, I thought of just moving that piece of code to makefile and calling it from here, just like other makefile tasks we already do that

Comment (conao3) [reply .github/workflows/lint.yml:37]: because current shell script depends `${{ github.sha }}` placeholder.

Comment (ericdallo) [reply .github/workflows/lint.yml:37]: Hum, got it
Ok, we can keep it here for now I think

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**
